diff --git a/includes/common.inc b/includes/common.inc
index c730d5f..dd288d6 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -2048,7 +2048,11 @@ function format_date($timestamp, $type = 'medium', $format = '', $timezone = NUL
   }
 
   // Create a DateTime object from the timestamp.
-  $date_time = date_create('@' . $timestamp);
+  if($timestamp != NULL && !empty($timestamp)){
+	$date_time = date_create('@' . intval($timestamp));
+  }else {
+	$date_time = date_create('@' . time());
+  }
   // Set the time zone for the DateTime object.
   date_timezone_set($date_time, $timezones[$timezone]);
 
